src/main/cpp/VideoState.h

```diff
@@ -78,6 +82,9 @@ extern "C" {
 
 #define MY_AV_TIME_BASE_Q av_make_q(1, AV_TIME_BASE)
 
+// fixed point to double
+#define CONV_FP(x) ((double) (x)) / (1 << 16)
```

Comment:
Not clear to me how this works...  Needs more explanation and maybe an example.
This obviously depends on the fixed point format chosen.

Reply:
This is used by std::hypot(double x, double y) function that computes the square root of the sum of the squares of x and y. In order to rotate the frame, we need from the AVStream a 3x3 transformation matrix describing an affine transformation that needs to be applied to the decoded video frames and numbers stored in the matrix are native endianness, as 16.16 fixed-point values. Converting matrix values to double is done through CONV_FP preprocessor in order to use hypot function.